src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_service_principals.py

```diff
@@ -0,0 +1,37 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+#pylint: skip-file
+import unittest
+import mock
+
+from azure.cli.core._util import CLIError
+from azure.cli.command_modules.acs.custom import (
+    _validate_service_principal,
+    _build_service_principal
+)
+
+
+class spClient:
+    def __init__(self, spMock, appMock):
+        self.service_principals=spMock
+        self.applications=appMock
+
+class AcsServicePrincipalTest(unittest.TestCase):
+    def test_validate_service_principal_ok(self):
+        spMock = mock.Mock()
+        spMock.list.return_value = []
+        client = spClient(spMock, None)
```

Comment:
I believe you don't need the `spClient`, just init an magicMock`, and assign the fields, like `service_principals`, with another magic mock object.